mobile/test/jni/jni_http_test_server.cc

```diff
@@ -0,0 +1,45 @@
+#include <jni.h>
+
+#include "test/common/integration/test_server.h"
+
+#include "extension_registry.h"
+#include "library/jni/jni_helper.h"
+#include "library/jni/jni_utility.h"
+
+// NOLINT(namespace-envoy)
+
+extern "C" JNIEXPORT jobject JNICALL
+Java_io_envoyproxy_envoymobile_engine_testing_HttpTestServer_start(JNIEnv* env, jclass, jint type,
+                                                                   jobject headers, jstring body) {
+  Envoy::JNI::JniHelper jni_helper(env);
+
+  Envoy::ExtensionRegistry::registerFactories();
+  Envoy::TestServer* test_server = new Envoy::TestServer();
+  test_server->startTestServer(static_cast<Envoy::TestServerType>(type));
+
+  auto cpp_headers = Envoy::JNI::javaMapToCppMap(jni_helper, headers);
+  auto cpp_body = Envoy::JNI::javaStringToCppString(jni_helper, body);
+  test_server->setResponse(cpp_headers, cpp_body);
+
+  auto java_http_server_class =
+      jni_helper.findClass("io/envoyproxy/envoymobile/engine/testing/HttpTestServer$HttpServer");
+  auto java_init_method_id =
+      jni_helper.getMethodId(java_http_server_class.get(), "<init>", "(JI)V");
+  int port = test_server->getServerPort();
+  return jni_helper
+      .newObject(java_http_server_class.get(), java_init_method_id,
+                 reinterpret_cast<jlong>(test_server), static_cast<jint>(port))
+      .release();
+}
+
+extern "C" JNIEXPORT void JNICALL
+Java_io_envoyproxy_envoymobile_engine_testing_HttpTestServer_00024HttpServer_shutdown(
```

Comment:
what does the 00024HttpServer mean in this?  I wasn't sure where that came from.

Reply:
ahh got it, thanks :)